staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -273,7 +273,9 @@ func (r *ScaleREST) Update(ctx context.Context, name string, objInfo rest.Update
 	if err = unstructured.SetNestedField(cr.Object, int64(scale.Spec.Replicas), strings.Split(specReplicasPath, ".")...); err != nil {
 		return nil, false, err
 	}
-	cr.SetResourceVersion(scale.ResourceVersion)
+	if len(scale.ResourceVersion) != 0 {
```

Comment:
why can it ever happen that `scale.ResourceVersion` is zero? This looks wrong.

Reply:
https://github.com/kubernetes/kubernetes/blob/564eff757cdd26a1322458876287298b8434b78b/staging/src/k8s.io/kubectl/pkg/scale/scale.go#L117-L119

because we don't get `Scale` resource before updating it now, so  the resource version can not be known in advance unless we specify it explicitly.

Reply:
If we supported unconditional PUT on CRs, this current code would be super dangerous!

* T1: I (e.g., HPA) send the scale command to scale to 5, with RV unset (RV = "")
* T2: The Get on line 240 happens
* T3: An unrelated party (e.g., the user) changes the spec.template.image or whatever
* T4: The update at line 280 happens

After this proposed change, the update returns a conflict instead of obeying the user (RV="" means unconditional PUT, no concurrency protection).

Prior to this change, if we supported unconditional PUT, the update would succeed, blowing away the change made at T3.

Both the existing code and this change are wrong. This needs to be done as a PATCH, not an update (or, if as an update, in a loop where conflict errors are handled). (The current code happens to not be problematic only because unconditional PUT is disallowed for CRs.)

If the user requests an unconditional PUT of scale, we should do that, but it may ONLY affect the scale field, we shouldn't destroy unrelated changes, nor should we return spurious conflicts for unrelated changes.